perceval/algorithm/tomography/tomography_mle.py

```diff
@@ -0,0 +1,400 @@
+from ..abstract_algorithm import AAlgorithm
+
+
+class TomographyMLE(AAlgorithm):
+    # Maximum Likelihood Estimation
+    def __init__(self, operator_processor: AProcessor, **kwargs):
+
+        super().__init__(processor=operator_processor, **kwargs)
+        self._nqubit, odd_modes = divmod(operator_processor.m, 2)
+        if odd_modes:
+            raise ValueError(
+                f"Input processor has an odd mode count ({operator_processor.m}) and thus, is not a logical gate")
+
+        if self._processor.is_remote:
+            raise TypeError("Tomography does not support Remote Processor yet")
+
+        self._gate_logical_perf = None
+
+        # following parameters are for the reconstruction algorithm
+        self.decelerate_factor = kwargs.get('decelerate_factor', 0.5)
+        # param decelerate_factor: Decreases the learning rate for a slower descent with a value in range (0,1)
+        self.init_learn_rate = kwargs.get('init_learn_rate', 1)
+        # param init_learn_rate: initial learning rate
+        self.max_iterations = kwargs.get('max_iterations', 100)
+        # :param max_iterations: maximum number of iterations
+        self.convergence_precision = kwargs.get('convergence_precision', 1e-10)
+        # precision for convergence of the algorithm
+
+    _LOGICAL0 = BasicState([1, 0])
+    _LOGICAL1 = BasicState([0, 1])
+
+    @abstractmethod
+    def _log_likelihood_func(self, *args, **kwargs) -> float:
+        pass
+
+    @abstractmethod
+    def _grad_log_likelihood_func(self, *args, **kwargs) -> np.ndarray:
+        pass
+
+    @abstractmethod
+    def _povm_data(self):
+        pass
+
+    def _collect_data(self, prep_state_indices, state_meas_indices) -> list:
+        # performs measurements on the output_state for given preparation and measurement state indices at each qubit
+
+        output_distribution, self._gate_logical_perf = _compute_probs(self, prep_state_indices, state_meas_indices,
+                                                                      denormalize=False)
+
+        B = []
+        for j in range(2**self._nqubit):
+            measurement_state = BasicState()
+            for m in range(self._nqubit - 1, -1, -1):
+                if (j // (2 ** m)) % 2 == 0:
+                    measurement_state *= self._LOGICAL0
+                else:
+                    measurement_state *= self._LOGICAL1
+
+            B.append(output_distribution[measurement_state] / (3 ** self._nqubit))
+        return B
+
+    def _povm_state(self) -> list:
+        # Gives a set of measurement states, so they form a set of informationally complete measurements.
+        # For 1 qubit, they are (order important) : |0>,|1>,|+>,|->,|i+>,|i->,
+        # These measurement states are eigenvectors of the tensor products of Pauli operators
+
+        d = 2 ** self._nqubit
+        P = []
+        B = [np.array([[1], [0]], dtype='complex_'), np.array([[0], [1]], dtype='complex_'),
```

Comment:
Don't we want this vector to come from the file perceval/components/_pauli.py ?

Reply:
moved